### PR TITLE
Dark mode implementation

### PR DESCRIPTION
**Pull request by LucianPetri**
Dark Mode Implementation Details Here:  #90

### CODE REVIEW

**Comment by k-nearest-neighbor** [comment]
This is a great start. There are a few areas that need to be addressed:

- UI regressions: Changes to light mode theme; Header functionality and appearance
- UI that isn't incorporated into dark mode yet / well.
- Aesthetics.

I'm going to go through this PR and highlight these, probably committing.

**Comment by fozziethebeat** [comment]
This is a pretty big PR touching a lot of files.  Let's hold off merging other react component changes until this is reviewed and we're comfortable with it (or deciding some other resolution strategy).

**Comment by AbdBarho** [comment]
I also agree that this is a great start. thank you @LucianPetri.

I would like to propose a strategy for moving forward with this:

Ideally, all dark mode handling should be done either be configuring the theme directly, or by having some conditional styling for the components under `/components` only. This should also include font-size, shadows, gaps, padding, etc...

Everything under `/pages` should only care about what is being shown, not its design, by using the aforementioned components.

This should make our work future proof, and changes to the design both easy and global.

I am not sure if browser support is there yet, or if chakra ui supports it, but [Container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Container_Queries) sound really promising for paddings margins and such...

File: website/src/pages/index.tsx
Comment (k-nearest-neighbor) [inline website/src/pages/index.tsx:24]: This debugging hack snuck in to my commit. Will be setting it back to session.

File: website/src/styles/Theme/index.ts
Comment (fozziethebeat) [inline website/src/styles/Theme/index.ts:2]: Rather than import each components theme here can we create a `website/src/styles/Theme/components/index.ts` file.

With that we can simplify this as 
```
import * as components from "./components";
```

And then down below we don't need to define a `const components = { ...}`.  We can just use the import

File: website/src/components/Loading/LoadingScreen.jsx
Comment (fozziethebeat) [inline website/src/components/Loading/LoadingScreen.jsx:4]: If you're just replacing `div` elements without the most similar chakra element, i'd suggest `Box` instead of `Container`

**Review by fozziethebeat — COMMENTED** [review]
This looks pretty close to being ready.  What's stopping us from handling the  merge conflicts and checking it in as is?

**Comment by fozziethebeat** [comment]
I took a look at this branch.  I think we should just check this in as is and then revise the small UI issues in later (tinier) PRs.

The key things to fix before approval:
- Fix the merge the conflicts above
- Make sure the action boxes in tasks all are visible.  Right now `rank_assistant_replies` is visible and looks great.  The others done (i'm guessing because of how that box is styled).  I think the fix is as simple as copy whatever is done in `rank_assistant_replies` to the other tasks.

**Comment by k-nearest-neighbor** [comment]
Alright. All checks passing and ready.

**Review by fozziethebeat — APPROVED** [review]
Great to finally get this in.

I'll make some small targeted fixes for things in a follow up PR

**Pull request merged**
